### PR TITLE
Create os reporting endpoint, models, views and test cases

### LINKED ISSUE #64: I want to obtain a report on the operating systems running in my scanned environment via a REST interface.
## Specify type:
- Enhancement

## Description:
Provide the ability to obtain an operating system report on a scanned environment.

**Method:** GET
**URI:** `api/v1/reports/?type=os`
**Response:** 200 
```
[
  {'fact_collection_id': <id>,
   'report':
      [
        {'os': <os_name>, 'count': <integer>},
        {'os': <os_name>, 'count': <integer>},
      ...
      ]
]
```

**Method:** GET
**URI:** `api/v1/reports/?fact_collection_id=<id>&type=os`
**Response:** 200 
```
[
   {'os': <os_name>, 'count': <integer>},
   {'os': <os_name>, 'count': <integer>},
 ...
]
```

___
_Work items:_

- [ ] https://github.com/quipucords/quipucords/issues/75

### LINKED ISSUE #75: Create os reporting endpoint, models, views and test cases
## Specify type:
- Enhancement
___

## Description:
Provide the ability to obtain an operating system report on a scanned environment.

**Method:** GET
**URI:** `api/v1/reports/?type=os`
**Response:** 200 
```
[
  {'fact_collection_id': <id>,
   'report':
      [
        {'os_release': <os_release>, 'count': <integer>},
        {'os_release': <os_release>, 'count': <integer>},
        ...
      ]
  }
]
```

**Method:** GET
**URI:** `api/v1/reports/?fact_collection_id=<id>&type=os`
**Response:** 200 
```
 {
   'fact_collection_id': <id>,
   'report':
      [
        {'os_release': <os_release>, 'count': <integer>},
        {'os_release': <os_release>, 'count': <integer>},
        ...
      ]
  }
```

_Validation:_
Check validation of fact collection identifier.

___
Associated with https://github.com/quipucords/quipucords/issues/64

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #75 
Closes #64

### CODE REVIEW

File: docs/swagger.yml [hunk 1]
```diff
@@ -136,6 +136,27 @@ paths:
...
+          in: "query"
+          description: "The Fact Collection ID to be queried"
+          type: "string"
+      responses:
+        200:
+          description: "List of reports for scanned systems"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/SystemReportOut"
```
Comment (chambridge) [inline docs/swagger.yml]: potential 404 condition here if the fact_collection_id isn't found

File: quipucords/api/report_model.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+"""Models to retrieve system reports."""
+
+from django.db import models
+from api.fact_model import FactCollection
+
+
+class SystemFingerprint(models.Model):
+    """Represents os installation count"""
+    fact_collection = models.ForeignKey(FactCollection,
```
Comment (chambridge) [inline quipucords/api/report_model.py]: should this be *fact_collection_id*?

File: quipucords/api/report_views.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+            # For each id, build a report and add to results array
+            for fact_collection_value in fact_collection_value_set:
+                fact_collection_id = fact_collection_value['fact_collection']
+                report = self.build_report(fact_collection_id)
+                if report is not None:
+                    collection_report_list.append(report)
+                else:
+                    logger.error(
+                        'fact_collection_id %s no longer exists',
```
Comment (chambridge) [inline quipucords/api/report_views.py]: This error really means that no System finger prints could be found associated with the given fact_collection_id (potentially due to some internal delete?).

File: quipucords/api/report_model.py (hunk 2 above)
Comment (kholdaway) [reply quipucords/api/report_model.py]: No because its the object not the id

File: docs/swagger.yml (hunk 1 above)
Comment (kholdaway) [reply docs/swagger.yml]: Added 404 to swagger

**Comment by kholdaway** [comment]
@chambridge Updates complete.  Take another look.

**Pull request merged**